### PR TITLE
core: Add Encrypted.Resize method

### PR DESCRIPTION
**Pull request by mvollmer**
To go with Filesystem.Resize.

### CODE REVIEW

**Comment by vpodzime** [comment]
Please add some tests to the *src/tests/dbus-tests/test_70_encrypted.py* file.

**Comment by pothos** [comment]
Just a small heads-up concerning the constant size of 512, if `crypt_resize` and thus `bd_crypto_luks_resize` directly support LUKS2, then things might break if the sector size is not 512:

http://www.saout.de/pipermail/dm-crypt/2017-December/005771.html
>   LUKS2 and plain crypt devices can be now configured with larger encryption
>   sector (typically 4096 bytes, sector size must be the power of two,
>   maximal sector size is 4096 bytes for portability).
>   Large sector size can decrease encryption overhead and can also help
>   with some specific crypto hardware accelerators that perform very
>   badly with 512 bytes sectors.
>   …
>   For LUKS2, the sector size is stored in metadata and cannot be changed later.

**Comment by mvollmer** [comment]
> things might break if the sector size is not 512:

How can we get the sector size?

**Comment by mvollmer** [comment]
Whoops, clicked ten pixels too far to the left.

**Comment by mvollmer** [comment]
> Please add some tests to the src/tests/dbus-tests/test_70_encrypted.py file.

Done.

**Comment by vojtechtrefny** [comment]
Jenkins, ok to test.

File: src/udiskslinuxencrypted.c [hunk 1]
```diff
@@ -803,10 +804,143 @@ handle_change_passphrase (UDisksEncrypted        *encrypted,
...
+                                                     object,
+                                                     action_id,
+                                                     options,
+                                                     message,
+                                                     invocation))
+    goto out;
+
+  job = udisks_daemon_launch_simple_job (daemon,
+                                         UDISKS_OBJECT (object),
+                                         "encrypted-resize",
```
Comment (vojtechtrefny) [inline src/udiskslinuxencrypted.c]: Please add this job type to `udisks/udisksclient.c` (function `udisks_client_get_job_description`, line 2476) and `data/org.freedesktop.UDisks2.xml` (`org.freedesktop.UDisks2.Job` interface description, line 2455).

Also please add the new generated functions to `doc/udisks2-sections.txt.in.in` (`UDisksEncrypted` section).

**Comment by vojtechtrefny** [comment]
We will definitely need to do some changes for LUKS 2 but it can wait. I did some changes in libblockdev to fix unlocking of LUKS 2 devices but resize needs more changes to work properly.

File: src/udiskslinuxencrypted.c (hunk 1 above)
Comment (mvollmer) [reply src/udiskslinuxencrypted.c]: Done.

Comment (mvollmer) [reply src/udiskslinuxencrypted.c]: I noticed this:

```
diff --git a/doc/udisks2-sections.txt.in.in b/doc/udisks2-sections.txt.in.in
index 772709fc..3aa7a1f0 100644
--- a/doc/udisks2-sections.txt.in.in
+++ b/doc/udisks2-sections.txt.in.in
@@ -912,7 +912,7 @@ udisks_filesystem_complete_unmount
 udisks_filesystem_call_resize
 udisks_filesystem_call_resize_finish
 udisks_filesystem_call_resize_sync
-disks_filesystem_complete_resize
+udisks_filesystem_complete_resize
 udisks_filesystem_call_check
 udisks_filesystem_call_check_finish
 udisks_filesystem_call_check_sync
```

**Comment by vojtechtrefny** [comment]
I've included this change (together with luks2 resize support) into #552 Thank you for the patch and sorry it took us that long.

**Pull request closed without merging**